week5/Calc.hs

```diff
@@ -0,0 +1,95 @@
+{-# OPTIONS_GHC -Wall #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE TypeSynonymInstances #-}
+
+module Calc where
+
+import ExprT
+import Parser
+import StackVM
+import Data.Maybe
+import qualified Data.Map as M
+
+-- Exercise 1
+eval :: ExprT -> Integer
+eval (ExprT.Lit x) = x
```

Comment:
You've imported `ExprT` module implicitly by writing just `import ExprT`. This means that all exported from this module functions and data types are brought into scope. So you can use just `Lit` name. However, using implicit imports is not a good practice because it's not clear which module brings what names into scope and it leads to name conflicts. So, according to our style guide, we either need to have explicit import lists (like `import ExprT (Exprt (..), ...)`) or `qualified` imports.